presto-testing/src/main/java/io/prestosql/testing/datatype/DataTypeTest.java

```diff
@@ -55,27 +72,66 @@ public void execute(QueryRunner prestoExecutor, Session session, DataSetup dataS
         List<Object> expectedResults = inputs.stream().map(Input::toPrestoQueryResult).collect(toList());
         try (TestTable testTable = dataSetup.setupTestTable(unmodifiableList(inputs))) {
             MaterializedResult materializedRows = prestoExecutor.execute(session, "SELECT * from " + testTable.getName());
-            assertEquals(materializedRows.getTypes(), expectedTypes);
-            List<Object> actualResults = getOnlyElement(materializedRows).getFields();
-            assertEquals(actualResults.size(), expectedResults.size(), "lists don't have the same size");
-            for (int i = 0; i < expectedResults.size(); i++) {
-                assertEquals(actualResults.get(i), expectedResults.get(i), "Element " + i);
+            checkResults(expectedTypes, expectedResults, materializedRows);
+            if (runSelectWithWhere) {
+                String prestoQuery = buildPrestoQueryWithWhereClauses(testTable);
+                MaterializedResult filteredRows = prestoExecutor.execute(session, prestoQuery);
+                checkResults(expectedTypes, expectedResults, filteredRows);
+            }
+        }
+    }
+
+    private String buildPrestoQueryWithWhereClauses(TestTable testTable)
+    {
+        List<String> predicates = new ArrayList<>();
+        for (int i = 0; i < inputs.size(); i++) {
+            Input<?> input = inputs.get(i);
+            if (input.isUseInWhereClause()) {
+                if (input.isNullValue()) {
+                    predicates.add(format("col_%d IS NULL", i));
+                }
```

Comment:
is not needed because `toPrestoLiteral` handles NULLs

Reply:
Is "IS NULL" and "IS NOT DISTINCT FROM NULL" the same? If I remove the null check at this level we will be forced to use the same comparison operator for NULL and non-NULL cases.

Reply:
> Is "IS NULL" and "IS NOT DISTINCT FROM NULL" the same?

yes